scripts/mvftoms.py

```diff
@@ -433,11 +433,14 @@ def _cp_index(a1, a2, pol):
 
                 # Select correlator products
                 # cp_index could be used above when the LazyIndexer
-                # supports advanced integer indices
-                vis_data = scan_data[:, :, cp_info.cp_index]
-
-                weight_data = scan_weight_data[:, :, cp_info.cp_index]
-                flag_data = scan_flag_data[:, :, cp_info.cp_index]
+                # supports advanced integer indices. Note that np.take is
+                # equivalent to but much faster than fancy indexing.
```

Comment:
Maybe it's because we are only selecting on one axis and `np.take` can somehow "take" advantage of that.

Reply:
Quite likely.